src/lib.rs

```diff
@@ -206,6 +210,36 @@ impl<R: Read> Iterator for MessageStream<R> {
     }
 }
 
+/// Trait for readers that can report their size.
```

Comment:
I'm not keen in introducing a new trait and increasing the API surface just for this use-case. IMO it would be better to have a `fn get_ref(&self) -> &R` method on `MessageStream` that just returns a reference to the reader c.f. https://docs.rs/flate2/latest/flate2/read/struct.GzDecoder.html#method.get_ref

Reply:
Yeah, that's fair.
I've update the PR accordingly.

Reply:
Great, thanks!